src/elm/Pages/Build/Logs.elm

```diff
@@ -385,9 +456,164 @@ defaultPosition =
     }
 
 
-{-| decodeAnsi : takes maybe log parses into ansi decoded log line array
+{-| decodeANSI : takes maybe log parses into ansi decoded log line array
+see: <https://package.elm-lang.org/packages/vito/elm-ansi>
+-}
+decodeANSI : String -> Array.Array Ansi.Log.Line
+decodeANSI log =
+    .lines <| Ansi.Log.update log defaultANSILogModel
+
+
+{-| processAndViewLine : takes Ansi.Log.Line and renders it into Html with ANSI styling
+see: <https://package.elm-lang.org/packages/vito/elm-ansi>
+this function has been modified to allow custom processing
+-}
+processAndViewLine : Ansi.Log.Line -> Html.Html msg
+processAndViewLine ( chunks, _ ) =
+    Html.div [] (List.foldl (\c l -> viewChunk c :: l) [ Html.text "\n" ] chunks)
+
+
+{-| viewChunk : takes Ansi.Log.Chunk and renders it into Html with ANSI styling
+see: <https://package.elm-lang.org/packages/vito/elm-ansi>
+this function has been modified to allow custom processing
+-}
+viewChunk : Ansi.Log.Chunk -> Html.Html msg
+viewChunk chunk =
+    Html.span (styleAttributes chunk.style) <|
+        parseLinksInChunk chunk
+
+
+{-| styleAttributes : takes Ansi.Log.Style and renders it into ANSI style Html attributes
+see: <https://package.elm-lang.org/packages/vito/elm-ansi>
+this function has been pulled in unmodified because elm-ansi does not expose it
+-}
+styleAttributes : Ansi.Log.Style -> List (Html.Attribute x)
+styleAttributes style =
```

Comment:
i'd rather not pull these in as static code, but the ansi module is too restricting and converts to html without much room for customization. 

however, the module's parsing behind the scenes is incredible, so i dont mind pulling these in.